refactor: step two theory rewrite and minor improvements (#61)
skills/integrate-mcp-with-copilot: .github/steps/2-step.md

We should probably look into them and start researching for other upgrades. Fortunately, with an MCP server for GitHub, triaging these and even doing some research to get ahead should be pretty quick! 🕵️

### 📖 Theory: How MCP Tool Calling Works in Agent Mode

Now that we have the GitHub MCP connected, let's look at how **agent mode** actually uses these external tools.

> With every prompt you send, Copilot also sends a catalog (list + schema) of all available tools to the LLM. The LLM plans and decides:

Reviewer: I would reword to not mention Copilot as a sort of forwarder to another LLM (even though that is really what it does).

```suggestion
With every prompt you send, Copilot also includes the catalog (list + schema) of available tools. Copilot can then plan and decide:
```

So, with the MCP Server available, we can now ask Copilot things beyond just our code. Here are some ideas to imagine the possibilities:
With every prompt you send, Copilot also sends a catalog (list + schema) of all available tools to the LLM. The LLM plans and decides:

> For example:

Reviewer: Why did you decided to remove all the examples?
Now it is only theory and you have to use your imagination of what the "GitHub MCP Server" thing enables.

Author: Well the way I did it is by removing the entire theory and asking copilot to generate one based on several blog posts/documentation links I linked and then I polished it.

Good catch that I removed the GitHub MCP specific examples, I'll add something about the GitHub MCP server specifically

Copilot then executes the chosen tool call(s) and streams results back to the LLM.

> ![Flowchart diagram illustrating how a user interacts with Copilot Agent Mode](https://github.blog/wp-content/uploads/2025/05/how-it-works.png)

Reviewer: Nice add! :copilot:
